07_errors/jimbotech/mapstore.go

```diff
@@ -0,0 +1,74 @@
+package main
+
+import (
+	"math/rand"
+)
+
+// MapStore stores puppies.
+type MapStore map[int32]*Puppy
+
+// length used for testing.
+func (s MapStore) length() int {
+	return len(s)
+}
+
+// CreatePuppy add a puppy to storage
+// but will modify the member ID.
+func (s MapStore) CreatePuppy(p *Puppy) (int32, error) {
+	if s == nil {
+		return 0, ErrNotConstructed
+	}
+	p.ID = rand.Int31()
```

Comment:
wat. i congratulate you on trying to leave land mines for the future developers. I curse you as a possible future developer.

Reply:
Previously I used uuid.New().ID() But unfortunately the uint will not turn into a int. This is in my view the next best thing. Similar in functionality. What is your concern? A repeat of the random number? The doco says: "Random numbers are generated by a Source. Top-level functions, such as Float64 and Int, use a default shared Source that produces a deterministic sequence of values each time a program is run." So I expect no repeats during during a run. Maybe I am misunderstanding your objection? What do you suggest?

Reply:
Added uniqueness check